src/core/HazelcastError.ts

```diff
@@ -63,6 +63,9 @@ export class HazelcastError extends Error {
         this.cause = cause;
         this.serverStackTrace = serverStackTrace;
         Error.captureStackTrace(this, HazelcastError);
+        if (cause !== undefined && cause !== null) {
```

Comment:
It should be possible to check against both `null` and `undefined` using:
```
if (cause != null) {
```

Reply:
As I "DM"ed you, we want to use === everywhere to get rid of some misuses of ==. There are a lot of places we use == where only null should be checked, however undefined is also allowed due to not using ===. 

Another possible thing to do is change errorFactory's createError so that it does not pass null to cause fields but passes undefined. Is this a back compatiblity problem, I am not sure. WDYT? @mdumandag